engine/common/splitter/engine.go

```diff
@@ -0,0 +1,163 @@
+package splitter
+
+import (
+	"errors"
+	"fmt"
+	"sync"
+
+	"github.com/onflow/flow-go/engine"
+	"github.com/onflow/flow-go/model/flow"
+	"github.com/onflow/flow-go/module"
+	"github.com/onflow/flow-go/network"
+	"github.com/rs/zerolog"
+)
+
+type Engine struct {
+	mu      sync.RWMutex
```

Comment:
Nit: something to indicate the coverage of the Mutex might help, e.g. `enginesMu`